specs/light_client/merkle_proofs.md

```diff
@@ -96,36 +133,43 @@ def chunk_count(typ: SSZType) -> int:
         return len(typ.get_fields())
     else:
         raise Exception(f"Type not supported: {typ}")
+```
 
-
-def get_item_position(typ: SSZType, index: Union[int, str]) -> Tuple[int, int, int]:
+```python
+def get_item_position(typ: SSZType, index_or_variable_name: Union[int, SSZVariableName]) -> Tuple[int, int, int]:
     """
-    Returns three variables: (i) the index of the chunk in which the given element of the item is
-    represented, (ii) the starting byte position within the chunk, (iii) the ending byte position within the chunk. For example for
-    a 6-item list of uint64 values, index=2 will return (0, 16, 24), index=5 will return (1, 8, 16)
+    Return three variables:
+        (i) the index of the chunk in which the given element of the item is represented;
+        (ii) the starting byte position within the chunk;
+        (iii) the ending byte position within the chunk.
+    For example: for a 6-item list of uint64 values, index=2 will return (0, 16, 24), index=5 will return (1, 8, 16)
     """
     if issubclass(typ, Elements):
+        index = int(index_or_variable_name)
         start = index * item_length(typ.elem_type)
         return start // 32, start % 32, start % 32 + item_length(typ.elem_type)
     elif issubclass(typ, Container):
-        return typ.get_field_names().index(index), 0, item_length(get_elem_type(typ, index))
+        variable_name = int(index_or_variable_name)
+        return typ.get_field_names().index(variable_name), 0, item_length(get_elem_type(typ, variable_name))
     else:
         raise Exception("Only lists/vectors/containers supported")
+```
 
-
-def get_generalized_index(typ: Type, path: List[Union[int, str]]) -> GeneralizedIndex:
+```python
+def get_generalized_index(typ: SSZType, path: Sequence[Union[int, SSZVariableName]]) -> Optional[GeneralizedIndex]:
     """
     Converts a path (eg. `[7, "foo", 3]` for `x[7].foo[3]`, `[12, "bar", "__len__"]` for
     `len(x[12].bar)`) into the generalized index representing its position in the Merkle tree.
     """
-    root = 1
+    root: Optional[GeneralizedIndex] = GeneralizedIndex(1)
     for p in path:
         assert not issubclass(typ, BasicValue)  # If we descend to a basic type, the path cannot continue further
         if p == '__len__':
             typ, root = uint64, root * 2 + 1 if issubclass(typ, (List, Bytes)) else None
```

Comment:
@vbuterin could you elaborate on why return `None`? Or should it just raise an exception?

Reply:
The idea here is that if you ask for the length of a fixed-length type, then that does not require knowing anything about the data, so there's no GeneralizedIndex that it makes sense to return.